src/PostgREST/QueryBuilder.hs

```diff
@@ -61,9 +61,22 @@ import           Data.Scientific         ( FPFormat (..)
 import           Prelude hiding          (unwords)
 import           PostgREST.ApiRequest    (PreferRepresentation (..))
 
+{-| The generic query result format used by API responses. The location header
+    is represented as a list of strings containing variable bindings like
+    @"k1=eq.42"@. If unused, it's null/Nothing rather than the empty list
+    because 'PostgreSQL.Binary.Decoder.arrayDimension' cannot decode an empty
+    array!
```

Comment:
It seems like you just need to upgrade your Hasql dependency bounds, because the latest releases have included [a fix for that](https://github.com/nikita-volkov/hasql/issues/46).